Feature/destination selection

MazyNoc opened the pull request:
Adding destination folder selection (drawable or mipmap) to Batch drawable importer.

MazyNoc committed:
Adding feature to select between Drawable and Mipmap destination folders

winterDroid commented on src/main/java/de/mprengemann/intellij/plugin/androidicons/dialogs/AddItemBatchScaleDialog.java:
```diff
@@ -299,6 +313,16 @@ private void updateFormat() {
+    private void updateDestination() {
+        destinationSpinner.removeActionListener(destinationListener);
+        destinationSpinner.removeAllItems();
+        for (Destination format : Destination.destinationValues()) {
```
just use `values`

winterDroid commented on src/main/java/de/mprengemann/intellij/plugin/androidicons/model/Destination.java:
```diff
@@ -0,0 +1,28 @@
+    Drawable,
+    Mipmap;
+
+    private static final Destination[] DESTINATION_VALUES = new Destination[] {
```
not necessary since you can use `values` directly

winterDroid commented on src/main/java/de/mprengemann/intellij/plugin/androidicons/model/Destination.java:
```diff
@@ -0,0 +1,28 @@
+
+    public static Destination from(String string) {
+        if (TextUtils.isEmpty(string)) {
+            throw new IllegalArgumentException();
```
please add a little description

winterDroid commented on src/main/java/de/mprengemann/intellij/plugin/androidicons/model/Destination.java:
```diff
@@ -0,0 +1,28 @@
+        if (TextUtils.isEmpty(string)) {
+            throw new IllegalArgumentException();
+        }
+        if (string.equalsIgnoreCase(Drawable.toString())) {
```
instead of going through all values, I'd prefer a loop like
```
for (Destination destination : values()) {
     if (destination.toString().equalsIgnoreCase(string)) {
          return destination;
     }
}
```

MazyNoc replied:
I was trying to mimic the code style of Format enum so that it is consistent across the codebase. 
Also, I could have uses the build-in enum.valueOf() but since it wasn't used in Format i didn't want to use it here either.
Should I move over to the build in function instead?

winterDroid replied:
Yeah I saw that, not sure why I did it back then like that. If you want you can change format as well ;)

MazyNoc replied:
valueof() - case sensitive, or from() with ignore case? From the usage i have not seen anything that wouldn't work with valueOf (except for exception handling)

winterDroid commented on src/main/java/de/mprengemann/intellij/plugin/androidicons/model/Destination.java:
```diff
@@ -0,0 +1,28 @@
+        } else if (string.equalsIgnoreCase(Mipmap.toString())) {
+            return Mipmap;
+        }
+        return null;
```
Are you sure it should be nullable? If so, please annotate the method with it or throw an `IllegalStateException` with a description mentioning the input string

winterDroid commented on src/main/java/de/mprengemann/intellij/plugin/androidicons/model/ImageInformation.java:
```diff
@@ -111,9 +114,14 @@ public Format getFormat() {
     public File getTargetFile() {
         return new File(String.format(TARGET_FILE_PATTERN,
                                       exportPath,
+                                      destination.toString().toLowerCase(),
```
either specify fixed `Locale` or use an argument for the actual folder name in the `Destination` constructor

winterDroid commented on src/main/java/de/mprengemann/intellij/plugin/androidicons/controllers/defaults/DefaultsController.java:
```diff
@@ -147,6 +162,8 @@ public void tearDown() {
         size = null;
         color = null;
         algorithm = null;
+        format = null;
```
oh good catch 👍

winterDroid commented on src/main/java/de/mprengemann/intellij/plugin/androidicons/controllers/settings/SettingsController.java:
```diff
@@ -234,6 +236,16 @@ public Format getFormat(Format defaultFormat) {
+
+    @Override
+    public Destination getDestination(Destination defaultDestination) {
+        return Destination.from(getOrDefault(DESTINATION,defaultDestination.toString()));
```
nit: space missing

winterDroid commented on src/main/java/de/mprengemann/intellij/plugin/androidicons/model/Destination.java:
```diff
@@ -0,0 +1,28 @@
+import de.mprengemann.intellij.plugin.androidicons.util.TextUtils;
+
+public enum Destination {
+    Drawable,
```
please make them all uppercase like `DRAWABLE`

winterDroid reviewed (changes requested):
Thanks for adding destination support. It would be great if you could squash the whole PR such that all the `merge` commits disappear.

MazyNoc committed:
changed from own implementation to valueOf() to get enum from string.
converted enums of Destination to uppercase
added folderName as constructor parameter for Destination enum do be able to separate the name from the constant (if ever necessary)

MazyNoc commented:
For the squash, I honestly don't know how to do that yet.

winterDroid commented:
Essentially you can just do `git rebase origin develop` to rebase your changes on top of the latest develop branch. Afterwards you need to force push.

MazyNoc commented:
ping! (in case you forgot this)

MazyNoc commented:
Any plans on releasing it to the IntelliJ repository?
